src/vs/vscode.d.ts

```diff
@@ -4091,6 +4091,63 @@ declare module 'vscode' {
 		 * [`globalState`](#ExtensionContext.globalState) to store key value data.
 		 */
 		storagePath: string | undefined;
+
+		/**
+		 * This extension's logger
+		 */
+		logger: Logger;
+	}
+
+	/**
+	 * The severity level of a log message
+	 */
+	export enum LogLevel {
+		Trace = 1,
+		Debug = 2,
+		Info = 3,
+		Warning = 4,
+		Error = 5,
+		Critical = 6,
+		Off = 7
+	}
+
+	/**
+	 * A logger for writing to an extension's log file, and accessing its dedicated log directory.
+	 */
+	export interface Logger {
+		readonly onDidChangeLogLevel: Event<LogLevel>;
+		readonly currentLevel: LogLevel;
```

Comment:
Why is the log level be exposed as API ?

Reply:
One use case might be when for when an extension is managing its own logging and writing to a file on its own.

Reply:
Has anyone asked for this? There are very many things we could add to the API, but IMHO it is a good practice to be as limited as possible and add something only if really needed...

Reply:
If we expose the log directory, we have to expose this, it makes no sense to only provide one.